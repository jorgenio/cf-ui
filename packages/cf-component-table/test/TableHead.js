import React from 'react';
import renderer from 'react-test-renderer';
import {
  TableHead,
  TableHeadUnstyled,
  TableRowUnstyled
} from '../../cf-component-table/src/index';
import felaTestContext from '../../../felaTestContext';
import { mount } from 'enzyme';

test('should render', () => {
  const component = renderer.create(
    felaTestContext(<TableHead>TableHead</TableHead>)
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('should render extra class name', () => {
  const component = renderer.create(
    felaTestContext(<TableHead className="extra">TableHead</TableHead>)
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('should pass down CSS query props', () => {
  const wrapper = mount(
    <table>
      <TableHeadUnstyled striped hover bordered condensed bare theadIndex={0}>
        <TableRowUnstyled />
        <TableRowUnstyled />
      </TableHeadUnstyled>
    </table>
  );

  expect(wrapper.childAt(0).childAt(1).prop('condensed')).toBeTruthy();
  expect(wrapper.childAt(0).childAt(1).prop('striped')).toBeTruthy();
  expect(wrapper.childAt(0).childAt(1).prop('bordered')).toBeTruthy();
  expect(wrapper.childAt(0).childAt(1).prop('hover')).toBeTruthy();
  expect(wrapper.childAt(0).childAt(1).prop('bare')).toBeTruthy();
  expect(wrapper.childAt(0).childAt(1).prop('theadIndex')).toBe(0);
  expect(wrapper.childAt(0).childAt(1).prop('rowIndex')).toBe(-1);
});
