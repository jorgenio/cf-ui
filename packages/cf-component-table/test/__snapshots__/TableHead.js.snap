// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<thead
  className="cf-1gxgayq"
>
  TableHead
</thead>
`;

exports[`should render extra class name 1`] = `
<thead
  className="extra"
>
  TableHead
</thead>
`;
